### PR TITLE
Stats tool merge

### PR DESCRIPTION
**Pull request by imrn99**
! This PR is the second out of the two that will make up the next release !
This PR contains the merge of the external tool used to compute data about the behavior of the main program, as well as all related changes to clean-up the repository. No modifications have been done to the main program's code beside documentation updates.

Main changes: 
- Repository has been reorganized with workspace in order to merge the external tool
  used for profiling.
- Merge of the external tool, `fastiron-stats`
- Update to existing documentation & tool documentation
- Update of all the profiling data to make use of more consistent CPU execution policies

TO-DO:
- Find a solution to the permissions issues created when using the scripts
- Analyze behavior of Quicksilver to make more in-depth comparisons
- Can the `f32` support be linked to a feature?

### CODE REVIEW

File: fastiron-stats/src/io_utils.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+        Vec::with_capacity(100),
+    ];
```
Comment (cedricchevalier19) [inline fastiron-stats/src/io_utils.rs:47]:
```suggestion
    let mut values: [Vec<f64>; N_TALLIED_DATA] = Default::default();
    values.iter_mut().for_each(|v| v.reserve(100));
```

File: fastiron-stats/src/processing.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    // The table is something like this
+    //
+    //                   | Source | Rr | Split
+    // PopulationControl | ...
+    // CycleSync         | ...
+    //
+    // gnuplot has the Y axis upside down, hence the vector:
+    vec![
+        correlation(
+            &tallies_data[TalliedData::Source as usize],
```
Comment (cedricchevalier19) [inline fastiron-stats/src/processing.rs:63]: Casting as `usize` should be avoided. Perhaps implementing `Index` for an ad-hoc type.

File: fastiron-stats/src/structures.rs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    Collision = 9,
+    Escape = 10,
+    Census = 11,
+    NumSeg = 12,
+    ScalarFlux = 13,
+    PopulationControl = 14,
+    CycleTracking = 15,
+    CycleSync = 16,
+}
+
```
Comment (cedricchevalier19) [inline fastiron-stats/src/structures.rs:33]: Perhaps this should be provided by `fastiron` to be sure that all is always correctly synchronized.

**Review by cedricchevalier19 — APPROVED** [review]
Perhaps adding a Changelog can improve readability.

**Comment by imrn99** [comment]
Stats tool merge

Main changes: 
- Repository has been reorganized with workspace in order to merge the external tool
  used for profiling.
- Merge of the external tool, `fastiron-stats`
- Update to existing documentation & tool documentation
- Update of all the profiling data to make use of more consistent CPU execution policies

TO-DO:
- Find a solution to the permissions issues created when using the scripts
- Analyze behavior of Quicksilver to make more in-depth comparisons
- Can the `f32` support be linked to a feature?

**Pull request merged**